feat: adding pull request template (#219)
BerkeleyLearnVerify/Scenic: .github/workflows/PULL_REQUEST_TEMPLATE.md

### Description
<!-- Provide a brief description of the changes introduced by this pull request -->

### Issue Link
<!-- Provide a link to the related issue on GitHub or another issue tracking system -->

### Checklist
- [ ] I have tested the changes locally via `pytest` and/or other means
- [ ] I have added or updated relevant documentation

> - [ ] I have auto formatted the code to black and istio conventions

Reviewer: ```suggestion
- [ ] I have autoformatted the code with black and isort
```

Author: Ack!
